File: JAVA/SelectionSort.java
// import java.lang.reflect.Array;

public class SelectionSort {

    public static void selectionSort(int arr[]){
        for (int i = 0; i < arr.length; i++) {
            // Taking minimum element every iteration
            int Min_idx = i;
            for (int j = i+1; j < arr.length; j++) {
                if (arr[j] < arr[Min_idx]) {
                    Min_idx = j;
                }
            }
            if(Min_idx != i){
                int temp = arr[Min_idx];
                arr[Min_idx] = arr[i];
                arr[i] = temp;
            }
        }
    }

    public static void main(String[] args) {
        int arr[] = {20,45,16,50,75,90};
        selectionSort(arr);
        System.out.println("Sorted array is :- ");
        // System.out.println(Array.toString(arr));
    }
}

File: Python/selection_sort.py
def selection_sort(arr):
    n = len(arr)
    comparisons = 0
    for i in range(n):
        min_idx = i
        for j in range(i+1, n):
            comparisons += 1
            if arr[j] < arr[min_idx]:
                min_idx = j
        arr[i], arr[min_idx] = arr[min_idx], arr[i]
    return arr, comparisons

# Example usage
array = [5, 2, 8, 1, 9, 3]
sorted_array, comparisons = selection_sort(array)
print("Sorted Array:", sorted_array)
print("Number of Comparisons:", comparisons)